src/common/host_device_vector.cu

```diff
@@ -358,19 +358,33 @@ struct HostDeviceVectorImpl {
     return data_h_;
   }
 
-  void Reshard(const GPUDistribution& distribution) {
+  void Shard(const GPUDistribution& distribution) {
     if (distribution_ == distribution) { return; }
-    CHECK(distribution_.IsEmpty() || distribution.IsEmpty());
-    if (distribution.IsEmpty()) {
-      LazySyncHost(GPUAccess::kWrite);
-    }
+    CHECK(distribution_.IsEmpty());
+    CHECK(!distribution.IsEmpty());
```

Comment:
The comparison on line 362 makes this check redundant (the check on like 363 should remain, however).

Reply:
Removed.